From f63caa8277de8243dbe717563e0fe4f374e57885 Mon Sep 17 00:00:00 2001
From: Eldar Abusalimov <eldar.abusalimov@jetbrains.com>
Date: Sat, 12 Aug 2017 16:00:57 +0300
Subject: [PATCH] pyconfig: Disable definition of hypot as _hypot

This fixes compilation errors on MinGW-W64:

    In file included from /usr/i686-w64-mingw32/include/c++/7.1.1/math.h:36:0,
                     from build-gnulib/import/math.h:27,
                     from /opt/include/python2.7/pyport.h:325,
                     from /opt/include/python2.7/Python.h:58,
                     from ../../gdb-8.0/gdb/python/python-internal.h:94,
                     from ../../gdb-8.0/gdb/python/py-block.c:24:
    /usr/i686-w64-mingw32/include/c++/7.1.1/cmath:1136:11: error: '::hypot' has not been declared
       using ::hypot;
               ^~~~~

Refer to https://stackoverflow.com/a/12124708/545027 for
the details.
---
 include/pyconfig.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/pyconfig.h b/include/pyconfig.h
index 5d1c90a..3bda86d 100644
--- a/include/pyconfig.h
+++ b/include/pyconfig.h
@@ -282,7 +282,8 @@ typedef int pid_t;
 #endif
 
 #define COMPILER "[gcc]"
-#define hypot _hypot
+/*https://stackoverflow.com/a/12124708/545027*/
+/*#define hypot _hypot*/
 #define PY_LONG_LONG long long
 #define PY_LLONG_MIN LLONG_MIN
 #define PY_LLONG_MAX LLONG_MAX
-- 
2.7.4

